make-go.bash

```diff
@@ -2,20 +2,36 @@
 
 set -e
 
+GO_SOURCE_URL="https://storage.googleapis.com/golang/"
+GO_BOOTSTRAP_SOURCE="go1.4.3.src.tar.gz"
+GO_BOOTSTRAP_SHA1="486db10dc571a55c8d795365070f66d343458c48"
+GO_SOURCE="go1.6.3.src.tar.gz"
+GO_SHA1="b487b9127afba37e6c62305165bf840758d6adaf"
+
 MYDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 export CGO_ENABLED=0
 
 if [ -z "$GOROOT_BOOTSTRAP" ]; then
     # We need Go 1.4 to bootstrap Go 1.5
     if [ -z $GOROOT ] || [[ $(go version) != go\ version\ go1.4* ]] ; then
-            git submodule update --init ext/golang/go1.4
-            # Build Go 1.4 for host
-            pushd ext/golang/go1.4/src
-            ./make.bash --no-clean
-            popd
-            # Add Go 1.4 to the environment
-            export GOROOT="$(pwd)/ext/golang/go1.4"
+      if [ ! -f go-1.4.tar.gz ]; then
+        wget -O go-1.4.tar.gz "$GO_SOURCE_URL$GO_BOOTSTRAP_SOURCE"
+      fi
+      sha1sum=$(sha1sum go-1.4.tar.gz | awk '{print $1}')
+      if [ "$GO_BOOTSTRAP_SHA1" != "$sha1sum" ]; then
+        echo "Error: invalid checksum"
+        exit 1
+      fi
```

Comment:
You could technically do the sha check after the wget, and trust that it's all good ;)